pkg/controllers/controller.go

```diff
@@ -41,10 +40,10 @@ func (c *GenericController) Reconcile(ctx context.Context, req reconcile.Request
 	// 1. Read Spec
 	resource := c.For()
 	if err := c.Get(ctx, req.NamespacedName, resource); err != nil {
-		if errors.IsNotFound(err) {
+		if errors.IsNotFound(err) || errors.IsGone(err) {
```

Comment:
What could cause the resource to be "Gone". I'm not aware of kubernetes using this. I also haven't seen it in other operator implementations.

Reply:
Jacob recently commented about this. Here's the doc that convinced me to include this.
https://kubernetes.io/docs/reference/using-api/api-concepts/#410-gone-responses

Reply:
BTW, I hadn't done all my homework so it's possibly unnecessary. According to the docs you found it looks like maybe not?


> When using resourceVersionMatch=NotOlderThan and limit is set, clients must handle HTTP 410 "Gone" responses. For example, the client might retry with a newer resourceVersion or fall back to resourceVersion="".

(I'm not sure if NotOlderThan and limit is actually being passed anywhere under the covers?)

Reply:
Necessarily we're using latest here, so Gone is not possible.

Reply:
Ok, seems worth reverting - apologies for any churn I may have caused here.